torch/_inductor/ir.py

```diff
@@ -3346,59 +3346,47 @@ def _prepare_convolution_fusion_create(
     function only supports the CPU device since conv post-op fusion kernel is only
     supported on CPU right now.
     """
-
-    x = cls.require_stride1(cls.realize_input(x))
-    weight = cls.require_stride1(cls.realize_input(weight))
-    assert x.get_device().type == "cpu" and weight.get_device().type == "cpu"
-    inputs = [x, weight]
     stride = tuple(stride_)
     padding = tuple(padding_)
     dilation = tuple(dilation_)
     assert isinstance(groups, int)
-    with FakeTensorMode():
-        output, *_ = cls.process_kernel(
-            torch.ops.aten.convolution,
-            x,
-            weight,
-            bias,
+    with torch._subclasses.FakeTensorMode():
+        x_fake = ir_node_to_tensor(x, guard_shape=True)
+        weight_fake = ir_node_to_tensor(weight, guard_shape=True)
+        bias_fake = (
+            ir_node_to_tensor(bias, guard_shape=True) if bias is not None else bias
+        )
+        output = torch.ops.aten.convolution(
+            x_fake,
+            weight_fake,
+            bias_fake,
             stride,
             padding,
             dilation,
             False,
             [0, 0],
             groups,
         )
+        req_stride_order = get_stride_order(output.stride())
 
-    output_size = output.shape
-    weight_shape = [
-        sympy.Integer(V.graph.sizevars.guard_static_shape(s)) for s in weight.get_size()
-    ]
-    _, _, *kernel_size = weight_shape
-    output_layout_str = (
-        "torch.contiguous_format" if output.is_contiguous() else "torch.channels_last"
-    )
-
-    if output_layout_str == "torch.channels_last":
-        stride_order = [0] + list(reversed(range(1, len(kernel_size) + 1)))
-        if len(stride_order) < len(output_size):
-            # add batch dim if it exists
-            stride_order = [len(stride_order)] + stride_order
-    else:
-        stride_order = list(reversed(range(len(output_size))))
+    x = cls.require_stride_order(x, req_stride_order)
+    weight = cls.require_stride_order(weight, req_stride_order)
```

Comment:
Should keep weight layout as in https://github.com/pytorch/pytorch/pull/88891?

Reply:
Yes, changed.